### PR TITLE
Added GetObject and PutObject methods and unit tests

### PR DESCRIPTION
**Pull request by Suruthi-G-K**

### CODE REVIEW

File: pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go [hunk 1]
```diff
@@ -1,14 +1,54 @@
...
+	"context"
+	"io/ioutil"
+	"net/http"
 	"testing"
 
+	"github.com/IBM/ibm-cos-sdk-go/aws/request"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3/s3iface"
+	"github.com/grafana/dskit/backoff"
+
```
Comment (adityacs) [inline pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go]: This line should be removed

File: pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go [hunk 2]
```diff
@@ -1,14 +1,54 @@
...
+	errMissingBucket = errors.New("bucket not found")
+	errMissingKey    = errors.New("key not found")
+	errMissingObject = errors.New("Object data not found")
+)
+
+type RoundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (f RoundTripperFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req)
+}
```
Comment (adityacs) [inline pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go]: This should be removed

**Review by adityacs — CHANGES_REQUESTED** [review]

File: pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go (hunk 1 above)
Comment (Suruthi-G-K) [reply pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go]: All these imports are used in the code @adityacs

Comment (adityacs) [reply pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go]: I mean, the empty line `14` should be removed

**Comment by tareqmamari** [comment]
@Suruthi-G-K @amankrsingh2000 could you rebase the PR?

**Review by tareqmamari — APPROVED** [review]

**Pull request merged**